include/measurement_kit/ooni/dns_injection.hpp

```diff
@@ -0,0 +1,41 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+#ifndef MEASUREMENT_KIT_OONI_DNS_INJECTION_HPP
+#define MEASUREMENT_KIT_OONI_DNS_INJECTION_HPP
+
+#include <measurement_kit/common.hpp>
+#include <measurement_kit/ooni/ooni_test.hpp>
+#include <measurement_kit/report.hpp>
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+void dns_injection(std::string, Settings, Callback<Var<Entry>>,
+                   Var<Reactor> = Reactor::global(),
+                   Var<Logger> = Logger::global());
```

Comment:
This is the real test, implemented by a function and easily testable independently of the rest.